packages/react-log-viewer/src/LogViewer/LogViewer.tsx

```diff
@@ -56,18 +59,53 @@ export const LogViewer: React.FunctionComponent<LogViewerProps> = memo(
     const [highlightedRowIndexes, setHighlightedRowIndexes] = useState<number[] | null>([]);
     const [currentSearchedItemCount, setCurrentSearchedItemCount] = useState<number>(0);
     const [parsedData, setParsedData] = useState<string[] | null>([]);
-    const loggerRef = React.createRef<any>();
+
+    const [currentWidth, setCurrentWidth] = useState<number | null>(width || 600);
+    const [loading, setLoading] = useState(true);
+    const [firstMount, setFirstMount] = useState(true);
+
+    const logViewerRef = React.createRef<any>();
+    const containerRef = React.createRef<any>();
+    let resizeTimer = null as any;
+
+    React.useEffect(() => {
+      if (containerRef && containerRef.current && loading) {
+        window.addEventListener('resize', callbackResize);
+        if (firstMount) {
+          setCurrentWidth((containerRef.current as HTMLDivElement).clientWidth);
+          setLoading(false);
+          setFirstMount(false);
+        }
+      }
+      return () => window.removeEventListener('resize', callbackResize);
+    }, [containerRef.current]);
+
+    const callbackResize = () => {
+      setLoading(true);
+      if (resizeTimer) {
+        clearTimeout(resizeTimer);
+      }
+      resizeTimer = setTimeout(() => {
+        setLoading(false);
+      }, 500);
+    };
+
+    useEffect(() => {
+      if (loading) {
```

Comment:
`!loading` seems to work a bit better testing locally, to make the width re-update after the load finishes instead of lagging behind the resize event (I noticed sometimes the width would update to a size captured before the resize event ends or between resize events if you move the mouse slowly, and not update after letting go of the mouse).